rudder-core/src/main/scala/com/normation/rudder/domain/queries/CmdbQuery.scala

```diff
@@ -453,14 +451,23 @@ case class JsonComparator(key:String, splitter:String = "", numericvalue:Boolean
     }
   }
 
-  override def buildRegex(attribute:String,value:String) : RegexFilter = {
-    val regexp = ".*%s.*".format(splitJson(attribute,value).getOrElse(Nil).mkString(".*"))
-    RegexFilter(key,regexp)
+  override def buildRegex(attribute:String,value:String) : Box[RegexFilter] = {
+    for {
+      splitted <- splitJson(attribute,value)
+      regexp = s".*${splitted.mkString(".*")}.*"
```

Comment:
%s was used used by the format function with "splitted.mkString(".*")"

Now it is replaced by ${splitted.mkString(".*")}